bin/varnishd/builtin.vcl

```diff
@@ -230,7 +230,10 @@ sub vcl_refresh_conditions {
 # We currently only revalidate 200 responses
 sub vcl_refresh_status {
 	if (obj_stale.status != 200) {
-		return (error(503, "Invalid object for refresh"));
+		# Can happen if VCL adds If-Modified-Since / If-None-Match
+		# for non-200 status objects. If deliberate, this sub can
+		# be overridden with sub vcl_refresh_status { return; }
+		return (error(503, "Invalid object for refresh (status)"));
```

Comment:
Again, I would change the error message to something more specific:

> Can only refresh for status code 200

And for the contents of the comment, I think they belong in either VCL design patterns in the sphinx docs, or the documentation for `vcl_refresh_status` if there's a dedicated section in a manual.

Reply:
You are right that this should be covered there. But I also think we should keep the comments in built-in VCL, simply because this is just more helpful to users. We already have some comments in the same direction and I believe we should continue to add such hints.